internal/updater/verify_test.go

```diff
@@ -37,3 +37,22 @@ func TestGPGVerify(t *testing.T) {
 	require.NoError(t, err)
 	assert.True(t, ok)
 }
+
+// TestGPGVerifyIn6Months tests that the signature is still valid in 6 months.
+// This is supposed to act as a canary so we don't forget to roll the key
+// before it expires. See README.md for details.
+func TestGPGVerifyIn6Months(t *testing.T) {
+	// Can not run parallel because we're overwriting the global timeNow variable.
+	timeNow = func() time.Time {
+		return time.Now().Add(6 * 30 * 24 * time.Hour)
+	}
+	defer func() {
+		timeNow = func() time.Time {
+			return time.Now()
+		}
+	}()
+
+	ok, err := gpgVerify(testData, testSignature)
+	require.NoError(t, err, "If TestGPGVerify succeeds but this test fails the self-updater key is about to expire. Please open an issue to update the key. Thank you.")
+	assert.True(t, ok)
```

Comment:
an alternative might be to just rely on:
`func (sig *Signature) SigExpired(currentTime time.Time) bool`
https://github.com/ProtonMail/go-crypto/blob/d703f494c90d2fa80fdbbaf33c084a7225411b55/openpgp/packet/signature.go#L760-L762
for the signature returned by `PrimarySelfSignature`:
https://github.com/ProtonMail/go-crypto/blob/d703f494c90d2fa80fdbbaf33c084a7225411b55/openpgp/keys.go#L887-L892


Something like:
```suggestion
// we only check the latest pubkey
keyList, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(pubkey[0]))
if err != nil || len(keyList) < 1 {
	debug.Log("failed to read public key: %q", err)
	return false, fmt.Errorf("failed to read public key: %w", err)
}

sig, _ := keyList[0].PrimarySelfSignature()
require.True(t, !sig.SigExpired(time.Now().AddDate(0,6,0)), "If TestGPGVerify succeeds but this test fails the self-updater key is about to expire. Please open an issue to update the key. Thank you.")
```

So that we don't need to change `gpgVerify` and use a global variable.

Reply:
Great suggestion. I'll try that.

Reply:
That didn't seem to work. The self-sig didn't have an expriation in my case. Something else must have been expired (subkey, signature, ...).